Bug Fix: emacs html export

aareman opened the pull request:
emacs html export requires the htmlize package which is no longer included in the latest orgmode for emacs (I'm using emacs 26.3). If you export to html -> emacs with a code block it breaks since it uses htmlize to syntax highlight. (btw, get the same error in emacs when trying to export)

That being said, when I install the htmlize package in emacs, exporting from emacs works, but still fails from orgmode.nvim. The reason is because the `--batch` flag doesn't source your emacs configuration and any packages you have. This PR adds a `--load` switch to the export command that points at your emacs config. This tells emacs running in batch mode to load a specific startup file, in this case the main emacs config.

> **NOTE** Changes beyond this. We may want to have a config option for specifying the location of an emacs config to use, since its not standard on all systems.

I know its failing tests/formatting. Just want to make sure this is an acceptable fix before polishing this PR. (See NOTE above for one outstanding issue)

kristijanhusak commented:
Didn't know about that, thanks. I guess this should also fix https://github.com/nvim-orgmode/orgmode/issues/87. 
Only issue that I see with this is that it's hardcoded. People may be using different path like `$HOME/.emacs` for example.
Lets do this:

* Introduce a property `emacs_config` in the `config/defaults.lua` file that should look something like this:
```lua
emacs_config = {
  executable_path = 'emacs',
  config_path = '$HOME/.emacs.d/init.el`
}
```
* Use the above config in the exporter 

That should solve everything, and make it configurable.

aareman commented:
@kristijanhusak I've made the updates and fixed the formatting issues. I also added basic documentation for it in the DOC.md file, Is there anywhere else I should put it?

kristijanhusak commented on lua/orgmode/export/init.lua:
```diff
@@ -111,11 +111,15 @@ function Export.emacs(format, extension)
   if vim.fn.executable('emacs') ~= 1 then
     return utils.echo_error('emacs executable not found. Make sure emacs is in $PATH.')
   end
+  local emacs = config.emacs_config.executable_path
```
Move this above to line 110, and include it in the check on line 111 instead of `'emacs'` string.

aareman replied:
Done

kristijanhusak commented:
@aareman looks ok, just address that one comment, and please squash commits into a single one. Thanks.

aareman commented:
I made the changes, and squashed the commits.

kristijanhusak commented:
Awesome, thanks!
